Source/RestKit/CodableExtensions.swift

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright IBM Corporation 2016-2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+import Foundation
+
+//===----------------------------------------------------------------------===//
+// Extensions: JSONEncoder
+//===----------------------------------------------------------------------===//
+
+public extension JSONEncoder {
+
+    /// Encode an optional object.
+    public func encodeIfPresent<T: Encodable>(_ value: T?) throws -> Data {
+        guard let value = value else { return Data() }
+        return try encode(value)
+    }
+
+}
+
+//===----------------------------------------------------------------------===//
+// Extensions: KeyedEncodingContainer<DynamicKeys>
+//===----------------------------------------------------------------------===//
+
+public extension KeyedEncodingContainer where Key == DynamicKeys {
```

Comment:
This is cool.  I hadn't thought about extending `KeyedEncodingContainer`.  I like this idea.